apps/arweave/src/ar_block_cache.erl

```diff
@@ -456,6 +456,27 @@ get_siblings(Tab, B) ->
 			)
 	end.
 
+update_timestamp(Tab, H, ReceiveTimestamp) ->
+	case ets:lookup(Tab, {block, H}) of
+		[{_, {B, Status, Timestamp, Children}}] ->
+			case B#block.receive_timestamp of
+				undefined ->
+					insert(Tab, {
```

Comment:
It's not thread-safe to update this table outside the ar_node_worker process.

Reply:
ah, oof. Okay I'll rework it

Reply:
I moved the update into ar_node_worker: https://github.com/ArweaveTeam/arweave/pull/581/commits/e6b2dd18b55d769aa62d2bfb187db946c5d3b48c